retrieval_service/app/app_test.py

```diff
@@ -783,3 +808,80 @@ def test_policies_search(m_datastore, app, method_name, params, mock_return, exp
     assert len(output) == params["top_k"]
     assert output == expected
     assert models.Policy.model_validate(output[0])
+
+
+@patch.object(id_token, "verify_oauth2_token")
+@patch.object(datastore, "create")
+def test_insert_ticket_missing_user_info(m_datastore, m_verify_oauth2_token, app):
+    m_datastore = AsyncMock()
+    m_verify_oauth2_token.side_effect = ValueError("invalid token")
+    with TestClient(app) as client:
+        response = client.post(
+            "/tickets/insert",
+            json={
+                "airline": "CY",
+                "flight_number": "888",
+                "departure_airport": "LAX",
+                "arrival_airport": "JFK",
+                "departure_time": "2024-01-01 08:08:08",
+                "arrival_time": "2024-01-01 08:08:08",
+            },
+        )
+        assert response.status_code == 422
```

Comment:
And is it possible to mock a success insert_ticket operation?

Reply:
> For this, it should be throwing a different error (similar to the list flight below):
> 
> ```
> raise HTTPException(
>             status_code=401,
>             detail="User login required for data insertion",
>         )
> ```
> 
> Will this work if we don't mock it at all? Since the header won't have the User-Id-Token field, so the retrieval of id_token will fail.

Thanks for pointing that out. Fixed!

Reply:
> And is it possible to mock a success insert_ticket operation?

That's a good idea. Added :)